pkg/controllers/management/multiclusterapp/state.go

```diff
@@ -32,92 +37,98 @@ func (m *MCAppStateController) sync(key string, mcapp *v3.MultiClusterApp) (runt
 	if mcapp == nil || mcapp.DeletionTimestamp != nil {
 		return mcapp, nil
 	}
+	state := active
 	if v3.MultiClusterAppConditionInstalled.IsUnknown(mcapp) && v3.MultiClusterAppConditionInstalled.GetMessage(mcapp) == "upgrading" {
-		return mcapp, nil
+		state = ""
```

Comment:
why do we reset the state?

Reply:
This is because when mcApp is upgrading, we don't want to calculate its state, will rename this to `mcappState`